maintenance/zero_downtime_migration.rst

```diff
@@ -0,0 +1,167 @@
+         --mongoCollection $COLLECTION_NAME \
+         --license $LICENSE_FILE \
+         --source empty_dir
+
+         rm -rf empty_dir
+
+   - Configure sharding for the entries collection using the command below:
+
+      .. code-block:: bash
+
+         DB_NAME=vonkdata
+         COLLECTION_NAME=vonkentries
+         CONNECTION_STRING=<YOUR_CONNECTION_STRING>
+
+         mongosh $CONNECTION_STRING <<EOF
+         sh.shardCollection("$DB_NAME.$COLLECTION_NAME", { type: 1, im: 1, cur: 1, cnt: 1, change: 1, res_id: "hashed" });
+         EOF
+
+3. Execute the FSI migration process:
+  
+   Initialize the migration by executing the following command:
+
+   .. code-block:: bash
+
+        RECOVERY_JOURNAL_DIRECTORY=./journal # A directory where the progress will be stored in case of a crash. FSI will quickly catch up to the place where an error occurred.
+        FHIR_VERSION=R4
+
+        LICENSE_FILE="<path to your license file>"
+
+        RUNNING_MODE=Continuous # Or AdHoc. If AdHoc mode is used, FSI will terminate when all the items from the old DB have been processed.
+
+        SOURCE_CONNECTION_STRING="<old db connection string>"
+        SOURCE_COLLECTION_NAME=vonkentries
+
+        CONNECTION_STRING="<new DB connection string>"
+        COLLECTION_NAME=vonkentries
+
+        dotnet fsi.dll \
+            --fhir-version $FHIR_VERSION \
+            --provisionTargetDatabase true \
+            --useRecoveryJournal $RECOVERY_JOURNAL_DIRECTORY \
+            --sourceType MongoDb \
+            --srcMongoCollection $SOURCE_COLLECTION_NAME \
+            --srcMongoConnectionString $SOURCE_CONNECTION_STRING \
+            --srcMongoRunningMode $RUNNING_MODE \
+            --update-existing-resources ErrorOnConflict \
+            --dbType MongoDb \
+            --mongoConnectionstring $CONNECTION_STRING \
+            --mongoCollection $COLLECTION_NAME \
+            --license $LICENSE_FILE
+
+   Migration completion is indicated by the following message: ``No new items found in the database. Waiting for 00:00:05 before retrying...``
+
+4. Provision the new version of Firely Server instance
+5. Update the reverse proxy configuration to direct traffic to the new Firely Server instance
+6. Upon successful migration verification, terminate the FSI migration tool and decommission the previous Firely Server installation.
+
+
+Verification Steps
+------------------
+
+After migration completion:
+
+1. Compare record counts between source and target databases:
+
+   .. code-block:: bash
+
+      SOURCE_CONNECTION_STRING="<old db connection string>"
+      CONNECTION_STRING="<new DB connection string>"
+
+      # For source database
+      mongosh $SOURCE_CONNECTION_STRING --eval "db.vonkentries.count()"
+      
+      # For target database
+      mongosh $CONNECTION_STRING --eval "db.vonkentries.count()"
+
+2. Verify data integrity by sampling records
+3. Test CRUD operations on the new Firely Server instance
```

Comment:
I would focus on Search for verification, and not alter any data with C,U and D.